### ISSUE TITLE
Mobile screen main menu issue

### ISSUE THREAD

**Issue by buildabeer** [buildabeer/buildabeer#47]
On a mobile sized screen, the length of the menu is not right. If you expand pieces of the menu, you are no longer able to scroll to the bottom of the menu.

**Comment by eyesurewould** [comment]
On small screens, the home screen needs to be completely re-thought. Scrolling is definitely a problem and spacing is a problem. It's not just the menu!

Please take a look at this [quick review](https://docs.google.com/spreadsheets/d/1kmb6u_4eD3U699UoEyoBJoAv8-oIM_50_bZ47fV7-sk/edit?usp=sharing) of all elements on the home screen and how they might be re-organized. Using this new plan should eliminate duplicate elements, reduce visual clutter, and keep the click paths lean for frequent tasks.

If you have click path analytics, we can verify that my assumptions (columns B and C) are correct, but I feel pretty confident about them.

Following this new pattern will also create a new place for Equipment which might inform the conversation on #25.

**Comment by buildabeer** [comment]
Notes as I'm going through your doc:
New Recipe does not require authentication (although saving does)
What is hero?
If browse/create recipes are removed from the top menu, there is no easy way to get to them from another page other than home, right?

I think a lot of the sheet makes sense.

**Comment by buildabeer** [comment]
OH, almost forgot, I have google analytics running - but I don't see anything about click path analysis. Is there a tool I should add for that? It definitely sounds extremely useful.

**Comment by eyesurewould** [comment]
Great catch about browse/create access from other pages. I recommended
having Browse on the home page and on the reference page, but I should have
included it  in the Top menu as well (fixed in the analysis). For Create I
recommended home page and top menu (thus available anywhere).

If I can't save a recipe, I probably would be frustrated if I got all the
way through making one and then found out I had to register / log in. Now,
if I clicked on Create and was told immediately that I'd need to
register/login, that's different (and if I missed that, my apologies).

Hero is typically the main visual focus of a page - top center and
typically large. It's a main component of Web 3.0 (versus web 2.0 which was
all about header, left nav, footer, and a busy content box). On
BuildA.beer's home page currently, the "Register here" could each be
considered the Hero (even though it's not at the top).

Regarding Gogle Analytics, this blog post offers some great ideas (but not
much depth - it's a starting point):
https://support.google.com/analytics/answer/1713056?hl=en
You probably want to look at Users Flow to understand the typical
navigation.

**Comment by buildabeer** [comment]
That seems reasonable. Does it say anything towards people trying it out? I
could see someone coming to try it, seeing they can't do anything without
registering and then leaving. Maybe I'm cynical though.

Interesting reads on web 3.0, it never ceases to amaze me how important ux
design is and how much goes into it. I'll bookmark this to look more into
the analytics this weekend as well.

**Comment by eyesurewould** [comment]
Registration is pretty painless (especially since there is not payment
required) so I don't think that is a big hurdle. Most people understand
that to generate content (like a recipe) platforms need you to authenticate
so the new content is associated with your account. As long as it's free,
most folks won't care - especially if there is a "Login with..." feature so
they can leverage an existing Facebook / Google / Github account.

(I personally never use "Login with...", but I know many folks love it and
I get the appeal)

**Comment by buildabeer-admin** [comment]
That sounds fair. I do think it would be good to have the 'login with' option, that seems pretty standard currently. What's your thought on email validation? I think I currently have that in place, but it takes a minute for the email to go through which I can see as annoying.

**Comment by eyesurewould** [comment]
I think email verification is essential, otherwise anyone can claim any
email address. Even if it's slow, it's necessary.

**Comment by eyesurewould** [comment]
Please take a look at this mock up and let me know if it works for you. The main goal here was to design for mobile first (which *should* make supporting mobile easier on the more complex pages).

https://build-a-beer-908b4.firebaseapp.com/

No worries if you hate it, this was just a quick stab at a re-design.

**Comment by buildabeer-admin** [comment]
Not too shabby! Something looks off about the recent recipe though. I'm not
sure exactly, but I think it is that it looks like it should be marking end
of page since it is full width, but then has more.

**Comment by eyesurewould** [comment]
Thanks.  Not sure I follow the problem. What was your viewport (mobile, tablet, desktop?).

It's a quick mock up so I didn't focus too much on content.

**Comment by buildabeer-admin** [comment]
Mobile

**Comment by buildabeer-admin** [comment]
I think it just feels like there should be something else before recent
recipes. I don't have any ideas on what though, haha..

**Comment by eyesurewould** [comment]
Ah, ok. Well, from the ux review, I identified 4 things for the center of the home page:
Register (when not logged in)
Login (when not logged in)
Create Recipe
Browse Recipes

I added "Recent Recipes" just as an idea for how else we can make the home page enticing. We can drop it if it doesn't work.

**Comment by buildabeer-admin** [comment]
I definitely do like having it there, I think it adds to the homepage which
I've always thought was lacking. I think my weird feeling goes away if the
buttons are stacked.
[image: image.png]

**Comment by eyesurewould** [comment]
Was there an image attached to that last comment? If so, it got removed...

**Comment by buildabeer-admin** [comment]
Yeah, that's what I get for responding through email I guess.
![image](https://user-images.githubusercontent.com/44954008/54625454-b7e5cb80-4a45-11e9-94ce-3bdf6be5d80d.png)

**Comment by eyesurewould** [comment]
Got it!

I'll proceed and let you know when I've got something to show.

**Comment by eyesurewould** [comment]
Please see #56  - I'm unable to test changes to the menu for authenticated users.

**Comment by buildabeer-admin** [comment]
Hopefully I've at least unblocked you w/ the pre-made users! I'll verify that the fix went through this evening as well.
